src/products/components/ProductVariantChannels/VariantDetailsChannelsAvailabilityCard/index.tsx

```diff
@@ -0,0 +1,61 @@
+import {
+  ProductVariantCreateDataQuery,
+  ProductVariantFragment,
+} from "@saleor/graphql";
+import React from "react";
+
+import { AvailabilityCard } from "./AvailabilityCard";
+import {
+  getAvailabilityCountForProduct,
+  getAvailabilityCountForVariant,
+} from "./availabilityCount";
+import { CardSkeleton } from "./CardSkeleton";
+import { CreateVariantTitle } from "./CreateVariantTitle";
+import { NotAvailable } from "./NotAvailable";
+
+interface VariantDetailsChannelsAvailabilityCardProps {
+  variant: ProductVariantCreateDataQuery["product"] | ProductVariantFragment;
+  onManageClick?: () => void;
+}
+
+const isProduct = (
+  x: ProductVariantCreateDataQuery["product"] | ProductVariantFragment,
+): x is ProductVariantCreateDataQuery["product"] =>
+  (x as ProductVariantCreateDataQuery["product"]).variants !== undefined;
+
+const VariantDetailsChannelsAvailabilityCard: React.FC<VariantDetailsChannelsAvailabilityCardProps> = ({
+  variant,
+  onManageClick,
+}) => {
+  if (!variant) {
+    return <CardSkeleton />;
+  }
+
+  const isAvailableInAnyChannels = !!variant.channelListings.length;
+
+  if (!isAvailableInAnyChannels) {
+    return <NotAvailable />;
+  }
+
+  if (isProduct(variant)) {
```

Comment:
If component and variable are named `variant...` then why would this be a product? 🤔